pkg/kubectl/cmd/get.go

```diff
@@ -182,13 +183,22 @@ func RunGet(f cmdutil.Factory, out, errOut io.Writer, cmd *cobra.Command, args [
 		return cmdutil.UsageError(cmd, usageString)
 	}
 
-	// always show resources when getting by name or filename
+	// determine if args contains "all"
+	for _, a := range args {
+		if a == "all" {
```

Comment:
> What is the motivation for showing all when printing all resource types? This is unrelated to the output type.

Although unrelated to the output type, I was thinking it would be a better user experience overall if `all` implied that even resources that are hidden by default would also be displayed.

> it looks like that code has the same issue. I am not sure if there is a great solution:

https://github.com/kubernetes/kubernetes/blob/6e5b455b7762f73f599dbec69c9ceb7cb4c8dfca/pkg/kubectl/resource/builder.go#L808

Would it be a valid assumption that if `all` comes before any other "kind", it can be implied that the user did mean "all resources", rather than `pod/all`? If this is the case, could we only check to see if `args[0] == "all"`, or would that be too fragile?